compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -1690,6 +1690,16 @@ impl<'hir> LoweringContext<'_, 'hir> {
         let yielded =
             opt_expr.as_ref().map(|x| self.lower_expr(x)).unwrap_or_else(|| self.expr_unit(span));
 
+        if !self.tcx.features().yield_expr() {
```

Comment:
```suggestion
        if !self.tcx.features().yield_expr() && !self.tcx.features().coroutines() {
```

to reduce the churn. It's fine for the coroutine feature to imply the yield_expr feature effectively

Reply:
My latest update enables `yield` with any of `yield_expr`, `coroutines`, or `gen_blocks`.